src/components/calcite-button/calcite-button.tsx

```diff
@@ -118,14 +127,20 @@ export class CalciteButton {
 
     const iconEndEl = (
       <calcite-icon
-        class="calcite-button--icon icon-end"
+        class={{ [CSS.icon]: true, [CSS.iconEnd]: true }}
         dir={dir}
         flipRtl={this.iconFlipRtl === "end" || this.iconFlipRtl === "both"}
         icon={this.iconEnd}
         scale={iconScale}
       />
     );
 
+    const contentEl = (
+      <div class={CSS.content}>
```

Comment:
I'm a little hesitant to make this a `div` - see the note about "permitted content" : https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button inside a button. Obviously that only applies when this is rendered as a button element, not an anchor, but I think we should still try and use a `span` if possible to adhere to that guideline.

Reply:
I'm cool with that. 
Do you know if this `span` selector here is still needed?

Reply:
sorry...meant to add this link.
https://github.com/Esri/calcite-components/blob/master/src/components/calcite-button/calcite-button.scss#L60